tests/test_yaml_file_types.py

```diff
@@ -0,0 +1,930 @@
+        ),
+        pytest.param(
+            INVALID_REPOSITORY_MULTIPLE_ERRORS,
+            pytest.raises(
+                SystemExit,
+                match=(
+                    r"Package name .* should be lowercase.*\n.*Package .* has invalid"
+                    r" version type"
+                ),
+            ),
+            id="invalid_repository_file_multiple_errors",
+        ),
+        pytest.param(
+            VALID_REPOSITORY_ADDITIONAL_UNKNOWN_PROPERTIES,
+            does_not_raise(),
+            id="valid_repository_file_additional_unknown_properties",
+        ),
+        pytest.param(
+            INVALID_REPOSITORY_ADDITIONAL_UNKNOWN_UPPERCASE_PROPERTIES,
+            pytest.raises(
+                SystemExit,
+                match=(
+                    r"property should be lowercase.*\n.*invalid property value type"
+                    r" \(2.2\).*"
+                ),
+            ),
+            id="invalid_repository_file_additional_unknown_uppercase_properties",
+        ),
+    ],
+)
+def test_repository_file_yaml_type(content, expectations):
+    with expectations:
+        RepositoryFile().from_yaml_string(content)
+
+
+LINT_MAINTAINER_TEST_REPO = """
+    zopfli:
+      "0.3":
+        source: pypi
+        make: pip
+        maintainer: scout
+        depends:
+          - setuptools
+          - python
+    setuptools:
+      68.0.0:
+        source: pypi
+        make: pip
+        maintainer: scout
+        depends:
+          - wheel
+          - python
+    python:
+      3.8.6:
+        make: sh
+        makefile: build__python-virtualenv.sh
+        maintainer: scout
+      '3.9':
+        maintainer: scout
+        makefile: build__python-virtualenv.sh
+        make: pip
+    wheel:
+      0.40.0:
+        make: pip
+        maintainer: scout
+"""
+
+
+@pytest.mark.parametrize(
+    "package_name, package_version, expectation, return_object",
+    [
+        pytest.param(
+            "python",
+            "3.8.6",
+            does_not_raise(),
+            _komodo_error("python", "3.8.6", "scout"),
+            id="lint_maintainer_returns_valid",
+        ),
+        pytest.param(
+            "pytest",
+            "3.9",
+            pytest.raises(KomodoException),
+            None,
+            id="lint_maintainer_throws_on_missing_package",
+        ),
+        pytest.param(
+            "python",
+            "4.0",
+            pytest.raises(KomodoException),
+            None,
+            id="lint_maintainer_throws_on_missing_package",
+        ),
+    ],
+)
+def test_repository_file_lint_maintainer(
+    package_name, package_version, expectation, return_object
+):
+    repo_file = RepositoryFile().from_yaml_string(LINT_MAINTAINER_TEST_REPO)
+    with expectation:
+        result = repo_file.lint_maintainer(package_name, package_version)
+    if return_object:
+        assert result == return_object
+
+
+VALID_PACKAGE_STATUS = (
```

Comment:
Strings can probably be inlined in pytest.param.

Reply:
This was to avoid having magic-strings in pytest.param, but it should probably be fine as all cases have descriptive ids. I also think some yaml-strings (VALID_REPOSITORY) are reused in some cases. Should I keep the reused ones as variables, and do all single-use ones inline?

Reply:
Although, I will keep the multiline yaml strings (e.g repository dict) as variables as they ruin readability.